app/uk/gov/hmrc/thirdpartydeveloperfrontend/domain/models/flows/Flow.scala

```diff
@@ -37,13 +38,23 @@ object FlowType extends Enum[FlowType] with PlayJsonEnum[FlowType] {
   case object NEW_APPLICATION_EMAIL_PREFERENCES extends FlowType
   case object NEW_APPLICATION_EMAIL_PREFERENCES_V2 extends FlowType
   case object GET_PRODUCTION_CREDENTIALS extends FlowType
+
+  def from[A <: Flow : TypeTag]: FlowType = {
+    typeOf[A] match {
+      case t if t =:= typeOf[EmailPreferencesFlowV2]                => FlowType.EMAIL_PREFERENCES_V2
```

Comment:
I'm sure there is a good reason to do it this way, but couldn't this just be written as 
case t: EmailPreferencesFlowV2 =>
etc?

Reply:
It's types rather than objects that happen to be of a type, so t is not an instance of EmailPreferencesFlowV2

Reply:
thanks